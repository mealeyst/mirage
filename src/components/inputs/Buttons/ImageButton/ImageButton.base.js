import React from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'

const CustomButton = ({arrow, className, children, ...props}) => {
  return (
    <div className={className}>
      <button {...props}>
        {children}
        {arrow && <span className='arrow'>&rarr;</span>}
      </button>
      <span className='borderBottom' />
    </div>
  )
}

const setWidth = css`
  ${props => `width: ${props.width}`};
`

const BaseImageButton = styled(CustomButton)`
  position: relative;
  display: flex;
  ${props => setWidth}
  overflow: hidden;

  > button {
    display: flex;
    align-items: center;
    justify-content: center;

    box-sizing: border-box;
    height: 30px;
    ${props => setWidth}
    padding: 0 1.5rem;

    text-align: center;

    letter-spacing: .1rem;
    ${props => props.sentanceCase
      ? 'text-transform: inherit;'
      : 'text-transform: uppercase;'
    }

    font-family: ${props => props.theme.fonts.primaryFont};
    font-size: 1.4rem;
    font-weight: 500;
    border: none;

    .arrow {
      margin-left: 1.2rem;
    }
  }

  > .borderBottom {
    display: block;
    height: .2rem;
    width: 100%;
    transform: translateX(-100%);
    transition transform 0.25s ease-in-out;
    position: absolute;
    left: 0;
    bottom: 0;
    z-index: 1;
  }

  :hover {
    > button {
      cursor: pointer;
    }
    > .borderBottom {
      transform: translateX(0);
      cursor: pointer;
    }
  }
`

BaseImageButton.propTypes = {
  arrow: PropTypes.bool,
  sentanceCase: PropTypes.bool,
  width: PropTypes.string
}

BaseImageButton.defaultProps = {
  width: '35rem'
}

/** @component */
export default BaseImageButton

export { CustomButton, setWidth }
