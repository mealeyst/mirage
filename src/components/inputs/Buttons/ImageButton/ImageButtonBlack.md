```js
<ImageButtonBlack>Click Me!</ImageButtonBlack>
```

## With Arrow
```js
<ImageButtonBlack arrow>Click Me!</ImageButtonBlack>
```
