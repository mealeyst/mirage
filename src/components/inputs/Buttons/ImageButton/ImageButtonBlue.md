```js
<ImageButtonBlue width={'350px'}>Click Me!</ImageButtonBlue>
```

## With Arrow
```js
<ImageButtonBlue arrow width={'350px'}>Click Me!</ImageButtonBlue>
```
