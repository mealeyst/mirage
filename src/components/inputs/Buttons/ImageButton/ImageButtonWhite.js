import PropTypes from 'prop-types'
import styled from 'styled-components'

import BaseImageButton from './ImageButton.base'

const ImageButtonWhite = styled(BaseImageButton)`
  > button {
    background-color: ${props => props.theme.colors.white};
    color: ${props => props.theme.colors.rocketBlue};
  }
  > .borderBottom {
    background-color: ${props => props.theme.colors.rocketBlue};
  }
`

ImageButtonWhite.propTypes = {
  theme: PropTypes.shape({
    colors: PropTypes.shape({
      rocketBlue: PropTypes.string,
      white: PropTypes.string
    })
  })
}

/** @component */
export default ImageButtonWhite
