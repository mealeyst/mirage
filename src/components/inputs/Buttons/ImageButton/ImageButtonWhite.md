```js
<div style={{backgroundColor: '#666', padding: '20px'}}>
  <ImageButtonWhite width={'350px'}>Click Me!</ImageButtonWhite>
</div>
```

## With Arrow
```js
<div style={{backgroundColor: '#666', padding: '20px'}}>
  <ImageButtonWhite arrow width={'350px'}>Click Me!</ImageButtonWhite>
</div>
```
