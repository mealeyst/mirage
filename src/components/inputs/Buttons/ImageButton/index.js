export {default as ImageButtonWhite} from './ImageButtonWhite'
export {default as ImageButtonBlue} from './ImageButtonBlue'
export {default as ImageButtonBlack} from './ImageButtonBlack'
