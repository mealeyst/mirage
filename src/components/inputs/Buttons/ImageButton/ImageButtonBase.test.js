import React from 'react'
import { css } from 'styled-components'
import 'jest-styled-components'

import ImageButton, { CustomButton, setWidth } from './ImageButton.base'

const { shallowWithTheme } = global

describe('(Styled Component) ImageButton', () => {
  const createImageButton = (props) => {
    return shallowWithTheme(<ImageButton {...props} />)
  }

  test('matching the snapshot', () => {
    expect(createImageButton())
    .toMatchSnapshot()
  })

  test('setting the width', () => {
    const width = '45rem'
    expect(createImageButton({width: width}))
    .toHaveStyleRule({
      width: css`${setWidth}`
    })
  })

  test('setting text to sentanceCase', () => {
    expect(createImageButton({sentanceCase: true}))
    .toHaveStyleRule({
      'text-transform': 'inherit'
    })
  })
})


describe('(Component) CustomButton', () => {
  const createCustomButton = (props) => {
    return shallowWithTheme(<CustomButton {...props} />)
  }

  test('matching the snapshot', () => {
    expect(createCustomButton())
    .toMatchSnapshot()
  })

  test('setting the className', () => {
    const className = 'example-class'
    expect(
      createCustomButton({className: className})
      .prop('className')
    ).toEqual(className)
  })

  test('setting the text of the button', () => {
    const text = 'Example Button'
    expect(
      createCustomButton({children: text})
      .find('button').text()
    ).toEqual(text)
  })

  test('setting the arrow prop', () => {
    expect(
      createCustomButton({arrow: true})
      .find('.arrow').length
    ).toEqual(1)
  })
})
