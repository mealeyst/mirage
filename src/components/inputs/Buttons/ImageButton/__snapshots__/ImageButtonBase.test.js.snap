// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) CustomButton matching the snapshot 1`] = `
<div>
  <button />
  <span
    className="borderBottom"
  />
</div>
`;

exports[`(Styled Component) ImageButton matching the snapshot 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 35rem;
  overflow: hidden;
}

.c0 > button {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: border-box;
  height: 30px;
  width: 35rem;
  padding: 0 1.5rem;
  text-align: center;
  -webkit-letter-spacing: .1rem;
  -moz-letter-spacing: .1rem;
  -ms-letter-spacing: .1rem;
  letter-spacing: .1rem;
  text-transform: uppercase;
  font-family: "ff-din-web",Helvetica,Arial,sans-serif;
  font-size: 1.4rem;
  font-weight: 500;
  border: none;
}

.c0 > button .arrow {
  margin-left: 1.2rem;
}

.c0 > .borderBottom {
  display: block;
  height: .2rem;
  width: 100%;
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
  -webkit-transition: -webkit-transform 0.25s ease-in-out;
  -webkit-transition: transform 0.25s ease-in-out;
  transition: transform 0.25s ease-in-out;
  position: absolute;
  left: 0;
  bottom: 0;
  z-index: 1;
}

.c0:hover > button {
  cursor: pointer;
}

.c0:hover > .borderBottom {
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  cursor: pointer;
}

<CustomButton
  className="c0"
  width="35rem"
/>
`;
